analyses/cell-type-ewings/exploratory_analysis/04-infercnv.Rmd

```diff
@@ -0,0 +1,480 @@
+    \(celltype_mat, celltype_method) {
+      ComplexHeatmap::Heatmap(
+        t(celltype_mat), # transpose because matrix rows are in common & we want a vertical arrangement
+        col = circlize::colorRamp2(c(0, 1), colors = c("white", "darkslateblue")),
+        border = TRUE,
+        ## Row parameters
+        cluster_rows = TRUE,
+        row_title = celltype_method,
+        row_title_gp = grid::gpar(fontsize = 12),
+        row_title_side = "left",
+        row_names_side = "left",
+        row_dend_side = "right",
+        row_names_gp = grid::gpar(fontsize = 10),
+        ## Column parameters
+        cluster_columns = FALSE,
+        column_title = "",
+        column_title_gp = grid::gpar(fontsize = 12),
+        column_names_side = "bottom",
+        column_names_gp = grid::gpar(fontsize = 10),
+        column_names_rot = 90,
+        ## Legend parameters
+        heatmap_legend_param = list(
+          title = "Jaccard index",
+          direction = "vertical",
+          legend_width = unit(1.5, "in")
+        ),
+        show_heatmap_legend = celltype_method == "SingleR",
+      )
+    }) |>
+  # concatenate vertically into HeatmapList object
+  purrr::reduce(ComplexHeatmap::`%v%`) |>
+  ComplexHeatmap::draw(
+    heatmap_legend_side = "right",
+    # add a margin to the heatmap so labels don't get cut off
+    padding = unit(c(2, 20, 2, 2), "mm")
+  )
+```
+
+This looks as we expect with mostly muscle cells lining up with tumor cells. 
+
+## Check for known CNVs in Ewing's 
+
+There are a few known copy number variations in Ewing's sarcoma: 
+
+- Gain of Chr8
+- Gain of Chr12
+- Gain of Chr1p
+- Loss of Chr16q
+
+Although these are the most frequent, there are patients who do not have any of these alterations and patients that only have some of these alterations. 
+See [Tirode et al.,](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC4264969/) and [Crompton et al.](https://doi.org/10.1158/2159-8290.CD-13-1037). 
+
+First we will compare the proportions across each chromosome between cells we have labeled as tumor or normal. 
+Then we will look specifically at the chromosomes commonly affected in Ewing sarcoma. 
+
+```{r}
+# prepare a data frame with one row per chromosome per cell for plotting
+proportion_df <- cnv_df |> 
+  # select only relevant columns for plotting
+  dplyr::select(barcodes, UMAP1, UMAP2, cnv_mean_classification, all_of(proportion_cols)) |> 
+  tidyr::pivot_longer(cols = all_of(proportion_cols),
+    names_to = "chromosome",
+    values_to = "proportion") |> 
+  # remove extra string in front of chr 
+  dplyr::mutate(
+    chromosome = stringr::str_remove(chromosome, "proportion_scaled_cnv_chr"),
+    # turn into numeric for easy ordering in plot 
+    chromosome = as.numeric(chromosome)
+  )
+
+```
+
+```{r, fig.height=10, fig.width=10}
+ggplot(proportion_df, aes(x = proportion, color = cnv_mean_classification)) +
+  geom_density() +
+  theme_bw() +
+  facet_wrap(vars(chromosome), ncol = 3)
+```
```

Comment:
A couple adjustments to the density plot here to make it more readable: 

One is to add bounds to the  density calculation, which should help make the plot not extend past (0,1), but you may still want to set the width of the plot as well with `coord_cartesian()`. I also set a bandwidth (which was just a guess, definitely see if you can adjust) to make the plots more comparable. The goal here is to make the plots less spikey than the default, with a secondary goal of making them more similar in the y-range. 

The other thing you could do is add scaling to the density plots so they all have a max of 1: `geom_density(aes(y = after_stat(scaled))`, which is not quite kosher, but does often make comparing shapes easier.

````suggestion
```{r, fig.height=10, fig.width=10}
ggplot(proportion_df, aes(x = proportion, color = cnv_mean_classification)) +
  geom_density(bounds = c(0,1), bw = 0.05) +
  theme_bw() +
  facet_wrap(vars(chromosome), ncol = 3)
```
````